router.go

```diff
@@ -102,6 +108,18 @@ func (r *defaultRouter) handleSession(sess Session, realmURI URI) {
 		}
 
 		log.Println(sess.Id, msg.MessageType(), msg)
+		if isAuthz, err := realm.Authorizator.IsAuthorized(msg, details); !isAuthz {
+			if err != nil {
+				log.Println(sess.Id, msg.MessageType(), err.Error())
```

Comment:
I know you haven't dealt with logging yet but this reminded me to remind you... You don't necessarily have to take a dependency on `logrus`.  You can do something similar to https://github.com/SpirentOrion/trace/blob/master/trace.go#L69-L71.  This will let the package default to the normal `log` but our code can pass in a `logrus` logger.  You'd lose the ability to add fields to the log message but at least the message will come out in the correct place.